src/us/kbase/workspace/database/provenance/ResolvablePID.java

```diff
@@ -0,0 +1,118 @@
+package us.kbase.workspace.database.provenance;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * A resolvable permanent identifier for a resource.
+ */
+public class ResolvablePID {
+
+	private final String id;
+	private final String description;
+	private final String relationshipType;
+
+	private ResolvablePID(final String id, final String description, final String relationshipType) {
+		this.id = id;
+		this.relationshipType = relationshipType;
+		this.description = description;
+	}
+
+	/**
+	 * Get the ID, for example DOI:10.25982/59912.37.
+	 * 
+	 * @return the id.
+	 */
+	public String getId() {
+		return id;
+	}
+
+	/**
+	 * Get a free text description of the ID.
+	 * 
+	 * @return the description, if present.
+	 */
+	public Optional<String> getDescription() {
+		return Optional.ofNullable(description);
+	}
+
+	/**
+	 * Get the relationship between the ID and the resource, for example
+	 * "isSupplementTo" or "isNewVersionOf".
```

Comment:
Actually I'm a bit confused by this. The class docs say that this class is an ID for a resource, but this seems to imply it's an ID for a resource that is somehow related to a *different* resource.

Reply:
Rewritten the docs so it should be a bit clearer.